docs(setup): add Grafana Log Dashboard Guide (#18331)
grafana/loki: docs/sources/setup/install/logging-guide.md

---
title: Grafana Log Dashboard Guide
menuTitle: Grafana Log Dashboard Guide
description: Step-by-step guide for setting up Grafana, Loki, and Grafana Alloy for log aggregation and visualization.
weight: 310
---

> This guide provides an official, end-to-end reference for deploying a unified log monitoring system using Grafana, Loki, and Grafana Alloy. Centralized log collection is essential for rapid troubleshooting, security auditing, and system performance analysis. By automating both log transfer and ingestion, this setup minimizes operational overhead and ensures that logs from multiple sources arrive in Grafana in near real time.

Reviewer: This is just one option for this installation workflow, not the "official" one.
```suggestion
This guide provides one example of an end-to-end reference for deploying a unified log monitoring system using Grafana, Loki, and Grafana Alloy. Centralized log collection is essential for rapid troubleshooting, security auditing, and system performance analysis. By automating both log transfer and ingestion, this setup minimizes operational overhead and ensures that logs from multiple sources arrive in Grafana in near real time.
```

---

## Step 1: Install Grafana via CLI

> Grafana is the visualization layer for metrics and logs. Installing via CLI guarantees reproducible deployments and supports automated provisioning in production environments.  

Reviewer: ```suggestion
Grafana is the visualization layer for metrics, logs, and traces. Installing via CLI guarantees reproducible deployments and supports automated provisioning in production environments.  
```

* For complete installation instructions on all supported platforms, see the official Grafana guide: [Grafana Installation](https://grafana.com/docs/grafana/latest/setup-grafana/installation/)  

> * This example uses Ubuntu/Debian; if you’re on another OS, adapt the commands below as needed.

Reviewer: ```suggestion
* This example uses Ubuntu/Debian; if you’re on another operating system, adapt the commands below as needed.
```

```bash

> sudo apt-get update

Reviewer: Curious as to why you didn't include the prerequisites from the [Grafana installation instructions](https://grafana.com/docs/grafana/latest/setup-grafana/installation/debian/) but started with what would have been step 5?

sudo apt-get install -y gnupg2 curl
curl https://packages.grafana.com/gpg.key | sudo apt-key add -
sudo add-apt-repository "deb https://packages.grafana.com/oss/deb stable main"
sudo apt-get update
sudo apt-get install -y grafana

> sudo systemctl start grafana-server

Reviewer: Looks like you might be missing a step here?
https://grafana.com/docs/grafana/latest/setup-grafana/start-restart-grafana/

I'm a bit concerned that you've condensed what seems to be pages and pages in the Grafana installation documentation into a single copy/paste block.

sudo systemctl enable grafana-server
````

> * Once installed, start Grafana and make sure it's accessible via a port (default: `3000`).

Reviewer: Unless you're providing instructions for modifying the default port, we'll assume connection via the default port.  Also, didn't we already start Grafana with the previous command?

```suggestion
* Once installed, start Grafana and make sure it's accessible via port`3000`.
```

> * After Grafana is running, create an admin account. Once logged in, you'll land on the Grafana homepage.

Reviewer: Do you mean just [logging in ](https://grafana.com/docs/grafana/latest/setup-grafana/sign-in-to-grafana/)here, which is by default as the `admin` account?
Or do you want people to [create a user account](https://grafana.com/docs/grafana/latest/administration/user-management/server-user-management/#add-a-user) for themselves and [assign it admin privileges](https://grafana.com/docs/grafana/latest/administration/user-management/server-user-management/assign-remove-server-admin-privileges/)?  (Again, I'm a bit worried that this topic is over simplifying what is a complex series of steps.)

---

## Step 2: Deploy Loki for Log Storage

Loki provides a scalable and cost-effective approach to storing and querying logs alongside your metrics data. Installing Loki next ensures that you have the backend log store ready for ingestion before configuring collection agents.

* For complete installation instructions on all supported platforms, see the official Loki guide: [Loki Installation](https://grafana.com/docs/loki/latest/setup/install/)

> * This example uses Docker/Docker Compose; if you’re using another approach, adapt the commands below as needed.

Reviewer: ```suggestion
* This example uses Docker/Docker Compose; if you’re using another approach, adapt the examples below as needed.
```

Also, it's helpful to give the name of the yaml file when you provide code examples, since there are so many different files (meaning, don't assume that users know where to copy/paste this).
